terraform/modules/hub/matomo.tf

```diff
@@ -0,0 +1,538 @@
+  name    = "analytics.${local.mgmt_domain}"
+  type    = "A"
+
+  alias {
+    name                   = aws_lb.matomo.dns_name
+    zone_id                = aws_lb.matomo.zone_id
+    evaluate_target_health = false
+  }
+}
+
+data "template_file" "matomo_task_def" {
+  template = file("${path.module}/files/matomo/matomo-task-def.json")
+
+  vars = {
+    nginx_image_identifier          = "${local.tools_account_ecr_url_prefix}-verify-nginx-tls@${var.nginx_image_digest}"
+    matomo_config_file_part_one_arn = aws_ssm_parameter.matomo_config_file_part_one.arn
+    matomo_config_file_part_two_arn = aws_ssm_parameter.matomo_config_file_part_two.arn
+    volume_name                     = local.volume_name
+    image_and_tag                   = "${local.tools_account_ecr_url_prefix}-verify-matomo@${var.matomo_image_digest}"
+    location_blocks_base64          = base64encode(file("${path.module}/files/matomo/nginx.conf"))
+    log_format_base64               = local.nginx_log_format_base64
+  }
+}
+
+resource "aws_ecs_task_definition" "matomo_task_def" {
+  family                   = "matomo"
+  container_definitions    = data.template_file.matomo_task_def.rendered
+  network_mode             = "awsvpc"
+  execution_role_arn       = aws_iam_role.matomo_execution.arn
+  requires_compatibilities = ["FARGATE"]
+  cpu                      = 512
+  memory                   = 2048
+
+  volume {
+    name = local.volume_name
+  }
+}
+
+data "template_file" "matomo_archiving_task_def" {
+  template = file("${path.module}/files/matomo/matomo-archiving-def.json")
+
+  vars = {
+    matomo_config_file_part_one_arn = aws_ssm_parameter.matomo_config_file_part_one.arn
+    matomo_config_file_part_two_arn = aws_ssm_parameter.matomo_config_file_part_two.arn
+    image_and_tag                   = "${local.tools_account_ecr_url_prefix}-verify-matomo@${var.matomo_image_digest}"
+    cronitor_url                    = var.matomo_archiving_cronitor_url
+  }
+}
+
+resource "aws_ecs_task_definition" "matomo_archiving_task_def" {
+  family                   = "matomo-archiving"
+  container_definitions    = data.template_file.matomo_archiving_task_def.rendered
+  network_mode             = "awsvpc"
+  execution_role_arn       = aws_iam_role.matomo_execution.arn
+  requires_compatibilities = ["FARGATE"]
+  cpu                      = 512
+  memory                   = 2048
+}
+
+data "template_file" "matomo_adhoc_task_def" {
+  template = file("${path.module}/files/matomo/matomo-adhoc-def.json")
+
+  vars = {
+    image_and_tag = "mysql:5"
+  }
+}
+
+resource "aws_ecs_task_definition" "matomo_adhoc_task_def" {
+  family                   = "matomo-adhoc" ## TODO: namespacing?
+  container_definitions    = data.template_file.matomo_adhoc_task_def.rendered
+  network_mode             = "awsvpc"
+  execution_role_arn       = aws_iam_role.matomo_execution.arn
+  requires_compatibilities = ["FARGATE"]
+  cpu                      = 512
+  memory                   = 2048
+}
```

Comment:
Matomo adhoc task is used for running SQL queries and extracting Matomo data from Matomo database.

How will performance analysts ssh in to the container inside Fargate?

Reply:
🤔 If this use case needs an EC2 instance, and can't be done in Fargate, I wonder if we can make it possible/easy to launch an ad hoc EC2 instance (spot, naturally)? Say using `gds verify matomo-console` or something

Reply:
Yes, this sounds good.

Reply:
we've agreed we don't need the ad hoc task in this PR; we'll capture a card to address the underlying need later.